components/script/dom/htmlinputelement.rs

```diff
@@ -1096,6 +1114,41 @@ impl Activatable for HTMLInputElement {
                                   EventBubbles::Bubbles,
                                   EventCancelable::NotCancelable);
             },
+            InputType::InputFile => {
+                let window = window_from_node(self);
+                let filemanager = window.resource_threads().sender();
+
+                let mut files: Vec<Root<File>> = vec![];
+
+                if self.Multiple() {
+                    let (chan, recv) = ipc::channel().expect("Error initializing channel");
+                    let msg = FileManagerThreadMsg::SelectFiles(chan);
+                    let _ = filemanager.send(msg).unwrap();
+
+                    match recv.recv().expect("IpcSender side error") {
+                        Ok(selected_files) => {
+                            for selected in selected_files {
+                                files.push(File::new_from_selected(window.r(), selected));
+                            }
+                        },
+                        // XXX: I don't think we should panic here, but how to handle better?
+                        Err(err) => panic!("Input file select error: {:?}", err)
```

Comment:
Treat it as if no files were selected. This will probably happen if file upload was canceled.